src/brainfuck/io/Io.java

```diff
@@ -0,0 +1,62 @@
+package brainfuck.io;
+
+import java.io.*;
+import java.util.Scanner;
+import brainfuck.exceptions.ReadInputException;
+
+/**
+ * Manage Applications's input and output flows
+ * 
+ * @author Nassim Bounouas
+ */
+public class Io{
+
+	/**
+	 * InputStream used as input flow
+	 */
+	private InputStream input = System.in;
+	
+	/**
+	 * OutputStream used as output flow
+	 */
+	private OutputStream output = System.out;
+
+	/**
+	 * Constructs an Io Object used to access in read or write
+	 */
+	public Io(String in, String out) {
+		try{	
+			if(in != null){
+				this.input = new FileInputStream(in);
+			}
+		
+			if(out != null){
+				this.output = new FileOutputStream(out);
+			}
+		}catch(FileNotFoundException e){
+			e.printStackTrace();
```

Comment:
Should throw another custom exception which will exit with error code 3. But only for the `FileInputStream`, the `FileOutputStream` will throw a regular `FileNotFoundException` if it cannot find the directory.